src/daemon/sigfetcher.rs

```diff
@@ -181,28 +200,32 @@ fn fetch_all_signatures(
 
     for (db_vault, mut db_txs) in vault_txs {
         let stk_keys = revaultd.stakeholders_xpubs_at(db_vault.derivation_index);
+        let our_stk_key = revaultd.our_stk_xpub_at(db_vault.derivation_index);
 
         for db_tx in &mut db_txs {
-            match db_tx.psbt {
-                RevaultTx::Unvault(ref mut unvault_tx) => {
-                    log::debug!("Fetching Unvault signature");
-                    get_sigs(&mut transport, &stk_keys, unvault_tx, &revaultd.secp_ctx)?;
+            match &db_tx.psbt {
+                RevaultTx::Unvault(_) => {
+                    log::debug!("Syncing Unvault signature");
                 }
-                RevaultTx::Cancel(ref mut cancel_tx) => {
-                    log::debug!("Fetching Cancel signature");
-                    get_sigs(&mut transport, &stk_keys, cancel_tx, &revaultd.secp_ctx)?;
+                RevaultTx::Cancel(_) => {
+                    log::debug!("Syncing Cancel signature");
                 }
-                RevaultTx::Emergency(ref mut emer_tx) => {
-                    log::debug!("Fetching Emergency signature");
+                RevaultTx::Emergency(_) => {
+                    log::debug!("Syncing Emergency signature");
                     debug_assert!(revaultd.is_stakeholder());
-                    get_sigs(&mut transport, &stk_keys, emer_tx, &revaultd.secp_ctx)?;
                 }
-                RevaultTx::UnvaultEmergency(ref mut unemer_tx) => {
-                    log::debug!("Fetching Unvault Emergency signature");
+                RevaultTx::UnvaultEmergency(_) => {
+                    log::debug!("Syncing Unvault Emergency signature");
```

Comment:
nit: maybe worth getting rid of the `match`?

```diff
diff --git a/src/daemon/database/bitcointx.rs b/src/daemon/database/bitcointx.rs
index 5b6df93..2f077f0 100644
--- a/src/daemon/database/bitcointx.rs
+++ b/src/daemon/database/bitcointx.rs
@@ -58,6 +58,15 @@ pub enum RevaultTx {
 }
 
 impl RevaultTx {
+    pub fn type_str(&self) -> String {
+        match self {
+            RevaultTx::Unvault(_) => "Unvault".to_string(),
+            RevaultTx::Cancel(_) => "Cancel".to_string(),
+            RevaultTx::Emergency(_) => "Emergency".to_string(),
+            RevaultTx::UnvaultEmergency(_) => "Unvault Emergency".to_string(),
+        }
+    }
+
     /// Serialize in the PSBT format
     pub fn ser(&self) -> Vec<u8> {
         match self {
diff --git a/src/daemon/sigfetcher.rs b/src/daemon/sigfetcher.rs
index 6380014..fefa85a 100644
--- a/src/daemon/sigfetcher.rs
+++ b/src/daemon/sigfetcher.rs
@@ -85,6 +85,8 @@ fn sync_sigs<C: secp256k1::Verification>(
     let mut contains_our_signature = false;
     let our_stk_key = our_stk_key.map(|k| k.key);
 
+    log::debug!("Syncing {} signature", tx.type_str());
+
     for (key, sig) in signatures {
         let pubkey = BitcoinPubKey {
             compressed: true,
@@ -203,22 +205,12 @@ fn fetch_all_signatures(
         let our_stk_key = revaultd.our_stk_xpub_at(db_vault.derivation_index);
 
         for db_tx in &mut db_txs {
-            match &db_tx.psbt {
-                RevaultTx::Unvault(_) => {
-                    log::debug!("Syncing Unvault signature");
-                }
-                RevaultTx::Cancel(_) => {
-                    log::debug!("Syncing Cancel signature");
-                }
-                RevaultTx::Emergency(_) => {
-                    log::debug!("Syncing Emergency signature");
-                    debug_assert!(revaultd.is_stakeholder());
-                }
-                RevaultTx::UnvaultEmergency(_) => {
-                    log::debug!("Syncing Unvault Emergency signature");
-                    debug_assert!(revaultd.is_stakeholder());
-                }
-            };
+            if matches!(
+                db_tx.psbt,
+                RevaultTx::Emergency(_) | RevaultTx::UnvaultEmergency(_)
+            ) {
+                assert!(revaultd.is_stakeholder())
+            }
             sync_sigs(
                 &mut transport,
                 &stk_keys,
```

Reply:
Cool :)